codesign/lib/src/file_codesign_visitor.dart

```diff
@@ -161,7 +204,7 @@ update these file paths accordingly.
     log.info('parsed binaries without entitlements are $fileWithEntitlements');
 
     // recursively visit extracted files
-    await visitDirectory(directory: parentDirectory, entitlementParentPath: artifactFilePath);
+    await visitDirectory(directory: parentDirectory, entitlementParentPath: "");
```

Comment:
What does an empty string mean?  Should `entitlementParentPath` be nullable instead?

Reply:
Right good question. Yeah it is kind of confusing, i have renamed `entitlementParentPath` to be `parentVirtualPath`.

Right, the purpose of this variable is to track the theoretical logical path of the file in the file system, assuming that none of the files have been moved or shuffled (either due to extraction or copying around). Previously I wrote a little example in go/code-signing-standalone: 
```
As we move along the file system, we sometimes deviate from the file tree and go to different folders. 
For example, we might extract current zip to another folder for processing, and switch into the other folder. 
This deviation makes it hard to compare our absolute file path with the file path listed in entitlements.txt, because our absolute file path is not accurate.
Instead, we maintain a `virtual file path` along the way, denoting the theoretical position in the filesystem had we not switched folders. 
This virtual path can be easily compared with the paths listed in entitlements.txt.

For example, in the picture above, even though we extract parent_1/parent_2/ios.zip to folder parent_1/folder_ios for processing, in the recursive function call, we will note the virtual file path as parent_1/parent_2/ios.zip. 
```

in other words, if we have a file a/b/c/d/e/f/g/file_binary, and assume while traversing the file system we extracted it and shuffled files around, so by the time we see file_binary it is now located at t/r/y/u/v/s/t/e/file_binary. This virtual path variable would serve as the source of truth that dictates the theoretical path. The virtual path variable would have been passed along and updated during the file system traversal and tells us the theorectical path of file_binary is a/b/c/d/e/f/g/file_binary.

And yeah I set the start path of virtual path to be an empty string. And it adds on filenames as it goes deeper into the file structure. For example, assume we have a file artifacts.zip/impellerc,  at the point of artifacts.zip the virtual path is empty string "", but at the point of impellerc the virtual path is updated to "" + "impellerc" which becomes "impellerc". As another example, assume the file is now artifacts.zip/Flutter/impellerc,  (1) at the point of artifacts.zip the virtual path is empty string "",  (2) at the point of  artifacts.zip/Flutter the virtual path is updated to "" + "Flutter" which becomes "Flutter", (3) at the point of artifacts.zip/Flutter/impellerc the virtual path is updated to "Flutter" + "impellerc" which becomes "Flutter/impellerc". 

So the concatenation of file paths is different based on whether we are starting a virtual path, or we are updating an existing virtual path. If we are starting with a virtual path, we don't want a front slash, so the virtual path of artifacts.zip/impellerc would be "impellerc", as opposed to "/impellerc".  If we are updating a virtual path, we don't want a slash in between, so the virtual path of artifacts.zip/Flutter/impellerc would be "Flutter/impellerc", as opposed to "Flutterimpellerc".